forui/lib/src/widgets/line_calendar.dart

```diff
@@ -0,0 +1,352 @@
+
+  /// The currently selected [LocalDate].
+  final ValueNotifier<LocalDate> selected;
+
+  /// The first date in this calendar carousel. Defaults to 1st January 2000.
+  final LocalDate epoch;
+
+  /// Today's date. Defaults to the [LocalDate.now].
+  final LocalDate today;
+
+  /// Creates a [FLineCalendar].
+  FLineCalendar({
+    required this.selected,
+    LocalDate? epoch,
+    LocalDate? today,
+    this.style,
+    super.key,
+  })  : epoch = epoch ?? _epoch,
+        today = today ?? LocalDate.now();
+
+  @override
+  State<FLineCalendar> createState() => _FLineCalendarState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('selected', selected))
+      ..add(DiagnosticsProperty('epoch', epoch))
+      ..add(DiagnosticsProperty('today', today));
+  }
+}
+
+class _FLineCalendarState extends State<FLineCalendar> {
+  late ScrollController _controller = ScrollController();
+
+  @override
+  void initState() {
+    widget.selected.addListener(() => setState(() {}));
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = widget.style ?? context.theme.lineCalendarStyle;
+
+    final textScalor = MediaQuery.textScalerOf(context);
+    final dateTextSize = textScalor.scale(style.unselectedDateTextStyle.fontSize!);
+    final dayTextSize = textScalor.scale(style.unselectedDayTextStyle.fontSize!);
+    final size = dateTextSize + dayTextSize + _textSpacing + (style.heightPadding * 2);
+
+    final offset = (widget.selected.value.difference(widget.epoch).inDays - 2) * size + style.itemPadding;
+    _controller = ScrollController(initialScrollOffset: offset);
+
+    return SizedBox(
+      height: size,
+      child: ListView.builder(
+        controller: _controller,
+        scrollDirection: Axis.horizontal,
+        padding: EdgeInsets.zero,
+        // TODO: calculate width of items based on the text font size.
+        itemExtent: size,
+        itemBuilder: (context, index) => Container(
+          padding: EdgeInsets.symmetric(horizontal: style.itemPadding),
+          child: _Tile(
+            style: widget.style,
+            selected: widget.selected,
+            date: widget.epoch.add(Duration(days: index)),
+            today: widget.today,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+}
+
+class _Tile extends StatelessWidget {
+  final FLineCalendarStyle? style;
+  final ValueNotifier<LocalDate> selected;
+  final LocalDate date;
+  final bool today;
+
+  const _Tile({required this.selected, required this.date, required LocalDate today, this.style})
+      : today = date == today;
+
+  Color _style(BuildContext context, bool selected) {
+    final style = this.style ?? context.theme.lineCalendarStyle;
```

Comment:
This can be omitted if the parent `LineCalendar`'s style in the `build(...)` method is passed in.